contracts/NFTStorefrontV2.cdc

```diff
@@ -293,6 +298,17 @@ pub contract NFTStorefrontV2 {
             return self.marketplacesCapability
         }
 
+        /// hasNFTPresentInListingProvider
+        /// Tells whether listed NFT is present in provided capability.
+        /// If it returns `false` then it means listing becomes ghost or sold out.
+        pub fun hasNFTPresentInListingProvider(): Bool {
+            if let providerRef = self.nftProviderCapability.borrow() {
```

Comment:
Really good question austin, from the Cadence docs:
```
The function returns nil when the targeted path is empty, i.e. nothing is stored under it. When the requested type exceeds what is allowed by the capability (or any interim capabilities), execution will abort with an error.
```

So I believe you are true. Thinking a little about how one can arrive to that scenario, I think we are fine. I mean, for being in trouble, you will have to move away your NFT and change the type under that path, then marketplaces won't be able to cleanup your ghost listing, and later you will have to restore the original type under the path and return the NFT there. Sounds like a really edgy case to me, I can live with that.

Another solution will be to do `self.nftProviderCapability.check()` before attempting to borrow, that won't cause the method to panic and will allow `cleanupGhostListings()` to execute. After writing down everything this might be the better solution actually ^^u

Reply:
@alilloig check can panic too :)

Reply:
> I don't understand the use case for ghost listing removal. Isn't NFTStorefrontV2 based on user has the storefront and marketplace manage via events ? I understand that listing can stay in user account, but when marketplace will go to clean up?

The use case is that a user might list their NFT for sale on two marketplaces at once to get more coverage on potential buyers. But what happens if they:
1. Sell on one of them
2. Transfer their nft elsewhere

In both cases, you have listings on marketplaces that are valid, and, should you ever find yourself in possession of that nft again, you would find yourself at risk of losing the NFT for a price that is (presumably) much lower than it is now worth. This happened a lot to users on OpenSea about a year ago, they had made listings that never expired and then in an effort to save gas on cancellations, moved their nfts to another wallet instead of cancelling. Once the NFT came back to their original wallet, it could get sniped for much lower than the floor price

Reply:
I do agree these edge cases are not a problem.